src/Service/RegisteredMethodManager.php

```diff
@@ -61,4 +62,33 @@ public function registerForMember(Member $member, MethodInterface $method, $data
 
         $this->extend('onRegisterMethod', $member, $method);
     }
+
+    /**
+     * Delete a registration for the given method from the given member, provided it exists. This will also remove a
+     * registered back-up method if it will leave the member with only the back-up method remaing
+     *
+     * @param Member&MemberExtension $member
+     * @param MethodInterface $method
+     * @return bool Returns false if the given method is not registered for the member
+     */
+    public function deleteFromMember(Member $member, MethodInterface $method): bool
+    {
+        $method = $this->getFromMember($member, $method);
+
+        if (!$method) {
+            return false;
+        }
+
+        $method->delete();
+
+        // If there is only one method remaining, and that's the configured "backup" method - then delete that too
+        if ($member->RegisteredMFAMethods()->count() === 1
+            && $member->RegisteredMFAMethods()->first()->MethodClassName
+                === Config::inst()->get(MethodRegistry::class, 'default_backup_method')
+        ) {
```

Comment:
I've been confused about this a lot recently.
I approve of this style (parenthesis and brace on new line), but PSR-2 decrees:

>There MUST NOT be a space before the closing parenthesis

for control structures.

What is the SilverStripe official stance on this, do we have an exception in our linting rules I've never noticed?
Because this is the only one I can see:
https://github.com/silverstripe/silverstripe-mfa/blob/27e22916a90209da257b772f4d512fe7f83fe712/phpcs.xml.dist#L8

Reply:
If you put the `{` on the same line then the control body looks like more conditions. I'd always thought PSR-2 forces you to drop the line

Reply:
Yeah, this is where I struggle. It looks stupid on the same line as you point out, but it is apparently non-compliant if dropped to a new line like this.

>## [5. Control Structures](https://www.php-fig.org/psr/psr-2/#5-control-structures)
>
>The general style rules for control structures are as follows:
>
> - There MUST be one space after the control structure keyword
> - There MUST NOT be a space after the opening parenthesis
> - There MUST NOT be a space before the closing parenthesis
> - There MUST be one space between the closing parenthesis and the opening brace
> - The structure body MUST be indented once
> - The closing brace MUST be on the next line after the body
>
>The body of each structure MUST be enclosed by braces. This standardizes how the structures look, and reduces the likelihood of introducing errors as new lines get added to the body.

Reply:
Technically [we're aiming for PSR-12](https://github.com/silverstripe/silverstripe-framework/issues/7991) for our style compatibility wherever possible now. [PSR-12 says this](https://github.com/php-fig/fig-standards/blob/master/proposed/extended-coding-style-guide.md):

>The closing parenthesis and opening brace MUST be placed together on their own line with one space between them.

It also says that the first condition must be on its own line.

If you want to get into some AST parsing you could look at contributing some improvements to the codesniffer PSR-12 ruleset to cover these =) the standard is not "accepted" yet, so there's likely to be cases like this that aren't in the codesniffer libraries yet.